File: Sources/YTools/Modules/System/SettingsModule.swift
import Foundation
import YToolsModuleKit

struct SettingsModule: YToolsModule {
    let descriptor = ModuleDescriptor(id: "settings", name: "设置")

    func search(_ request: ModuleSearchRequest) async throws -> [LauncherResult] {
        let term = request.query.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
        let keywords = ["settings", "preferences", "设置", "偏好设置"]
        guard !term.isEmpty, keywords.contains(where: { $0.hasPrefix(term) || term.hasPrefix($0) }) else {
            return []
        }
        return [LauncherResult(
            id: "settings:open",
            moduleID: descriptor.id,
            title: "YTools 设置",
            subtitle: "外观、快捷键、剪贴板和隐私",
            icon: .system("gearshape.fill"),
            score: 880,
            action: .openSettings
        )]
    }
}

File: src/YTools.Windows/Services/Modules/SettingsModule.cs
using YTools.ModuleKit;

namespace YTools.Services.Modules;

public sealed class SettingsModule : IYToolsModule
{
    public ModuleDescriptor Descriptor { get; } = new("settings", "设置");

    public Task<IReadOnlyList<LauncherResult>> SearchAsync(ModuleSearchRequest request)
    {
        var term = request.Query.Trim().ToLowerInvariant();
        var keywords = new[] { "settings", "preferences", "设置", "偏好设置" };
        if (string.IsNullOrEmpty(term) || !keywords.Any(keyword => keyword.StartsWith(term) || term.StartsWith(keyword)))
        {
            return Task.FromResult<IReadOnlyList<LauncherResult>>([]);
        }

        return Task.FromResult<IReadOnlyList<LauncherResult>>(
        [
            new LauncherResult(
                "settings:open",
                Descriptor.Id,
                "YTools 设置",
                "外观、快捷键、剪贴板和隐私",
                new ResultIcon.System("gearshape.fill"),
                880,
                new ResultAction.OpenSettings())
        ]);
    }
}
